### DISCUSSION TITLE
Does dagrun_timeout apply to non-scheduled DAGRuns?

### DISCUSSION THREAD

**Discussion by seifertm** [apache/airflow#28404] [Q&A] [answered]
The documentation for the DAG constructor clearly states that the `dagrun_timeout` argument only applies to scheduled DAGs. [0] However, the following DAG times out when triggered manually on a CeleryExecutor, even though it is not scheduled:
```
from datetime import datetime, timedelta

from airflow import DAG
from airflow.operators.bash import BashOperator


with DAG(
    "timeout_test",
    schedule_interval=None,
    start_date=datetime(2022, 12, 1),
    catchup=False,
    dagrun_timeout=timedelta(seconds=10),
    is_paused_upon_creation=False,
) as dag:
    do_nothing = BashOperator(
        task_id="sleep",
        bash_command="sleep 600",
    )

do_nothing
```

`SchedulerJob._do_schedule` [fetches all running DAGs](https://github.com/apache/airflow/blob/2.5.0/airflow/jobs/scheduler_job.py#L960) and checks against any configured dagrun_timeout in [_schedule_dag_run](https://github.com/apache/airflow/blob/fa2bec042995004f45b914dd1d66b466ccced410/airflow/jobs/scheduler_job.py#L1280-L1284). My understanding is that the timeouts of all running DAGs are checked, not just the scheduled ones.

Does that mean that either
(1) `dagrun_timeout` applies to all DAG runs (not just scheduled ones) and the documentation is incorrect?
(2) `dagrun_timeout` applies to scheduled DAG runs only. It is not guaranteed that non-scheduled DAG runs are cancelled when the timeout expires. It just happens to be an implementation detail that they are cancelled.

Can you shed some light onto the topic?

[0] https://airflow.apache.org/docs/apache-airflow/2.5.0/_api/airflow/models/dag/index.html#airflow.models.dag.DAG

**Comment by potiuk** [comment, marked as answer]
Both documentation and code align. I think you misunderstand what "scheduled", Scheduled means that scheduler planned executing them. When you trigger a dag run manually, scheduler is scheduling it to run "now". 

If you think you can formulate it better in the docs - PRs are always most welcome.

**Reply by seifertm** [reply]
Thanks! You're right that my question was based on a misconception of what a "scheduled" DAG run is.
Your explanation makes sense.
